### DISCUSSION TITLE
How to use n1spindown/n2spindown

### DISCUSSION THREAD

**Discussion by Octal450** [JSBSim-Team/jsbsim#997] [Support requests]
Hi,
I am trying to adjust the shutdown speed of the FGTurbine, and are using FlightGear v Next (2020.4).

However, these tags seem to have no effect on the turbine. No matter how low I set them, the turbine still shuts down at a wrong speed. I'm not sure if I am doing something wrong?

A weird thing also, with these tags in place and smaller values set, the "set-running" property no longer works. The engine blips in RPM and then shuts back down.

The tags I am using are `<n1spindown>` and `<n2spindown>`. The spinup versions work perfectly.

```xml
	<milthrust>60690</milthrust>
	<tsfc>0.544</tsfc>
	<bleed>0</bleed> <!-- Set by pneumatic system -->
	<ignitionn1>2.8</ignitionn1>
	<ignitionn2>29.4</ignitionn2>
	<n1spinup>0.091</n1spinup>
	<n2spinup>0.958</n2spinup>
	<n1spindown>0.01</n1spindown>
	<n2spindown>0.02</n2spindown>
	<n1startrate>0.578</n1startrate>
	<n2startrate>1.193</n2startrate>
	<idlen1>23.7</idlen1>
	<idlen2>62.9</idlen2>
	<maxn1>117.5</maxn1>
	<maxn2>112.5</maxn2>
	<augmented>0</augmented>
	<injected>0</injected>
```

Kind Regards,
Josh

**Comment by seanmcleod** [comment]
How did you come up with your very small spin down values? 

So taking a look, these are the default values if you don't specify any.

https://github.com/JSBSim-Team/jsbsim/blob/056bb71fc432681cb72b84fd2a68d79360bb1064/src/models/propulsion/FGTurbine.cpp#L74-L75

And they're used in the call to `Seek()`.

https://github.com/JSBSim-Team/jsbsim/blob/056bb71fc432681cb72b84fd2a68d79360bb1064/src/models/propulsion/FGTurbine.cpp#L172-L184

And this is the implementation of `Seek()`.

https://github.com/JSBSim-Team/jsbsim/blob/056bb71fc432681cb72b84fd2a68d79360bb1064/src/models/propulsion/FGTurbine.cpp#L400-L410

Note how the deceleration factor is calculated when calling `Seek()` - `N1/N1_spindown`, so given your very small spin down values I would expect the engine to spin down faster than normal, i.e. when you don't specify a spin down value.

Now I noticed in the `SpinUp()` function the spin up values aren't used as a denominator. 

https://github.com/JSBSim-Team/jsbsim/blob/056bb71fc432681cb72b84fd2a68d79360bb1064/src/models/propulsion/FGTurbine.cpp#L273-L278

**Comment by Octal450** [comment]
Hi Sean,
I completely was unaware that this was denominator... I'll admit I was ignorant and didn't check the code. I assumed it worked like the spinup values.

There doesn't seem to be any documentation on the thing: https://jsbsim.sourceforge.net/JSBSim/classJSBSim_1_1FGTurbine.html hence how I missed it.

I will recalculate everything and try again tomorrow.

Thank you!
Josh

**Comment by seanmcleod** [comment]
I guess we should also double-check why the spinup and spindown values are used differently and whether they shouldn't be consistent.

**Comment by Octal450** [comment]
Since it's been so many years since implemented (2018) I think we probably should just leave it how it is and just document that it works that way.

Kind Regards,
Josh

**Comment by Octal450** [comment]
You know what - it's actually better like this. That way the shutdown curve is somewhat realistic instead of linear.

Kind Regards,
Josh

**Reply by seanmcleod** [reply]
Forgetting about backwards compatibility for now, but doesn't this also apply to the spin up case, i.e. it should also be used as a denominator?

**Reply by Octal450** [reply]
No, spin up is driven by the starter either air or electric etc. it’s a fairly linear rise. Back compat is simple, use either or add a new prop. Not that anyways has used it that I’m aware of anyways since it’s undocumented. 

Kind Regards,
Josh

**Comment by Octal450** [comment]
One more thing - would it be possible to have this value on a table lookup? So that my panel state loading system can override the value with a very fast value for quickly shutting down the engine - as the real takes long. For startup, set-running works fine but I guess also having spinup and startrate on a table might be useful.

Kind Regards,
Josh
